FileIO/CMakeLists.txt

```diff
@@ -1,17 +1,10 @@
-set(SOURCES
-    CsvInterface.h
-    CsvInterface.cpp
-    GMSInterface.h
-    GMSInterface.cpp
-    GMSHInterface.h
-    GMSHInterface.cpp
-    PetrelInterface.h
-    PetrelInterface.cpp
-    TetGenInterface.h
-    TetGenInterface.cpp
-)
-if(Shapelib_FOUND)
-    list(APPEND SOURCES SHPInterface.h SHPInterface.cpp)
+GET_SOURCE_FILES(SOURCES_FileIO)
+set(SOURCES ${SOURCES_FileIO})
+if(NOT Shapelib_FOUND)
+    list(REMOVE_ITEM SOURCES SHPInterface.h SHPInterface.cpp)
```

Comment:
Would it be a better solution to just #ifdef out everything in these two files if shapelib is not found rather than removing the files here?